File: Chapter1(Node_and_Topics)/indigo/src/minimal_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Float64.h>

void callback(const std_msgs::Float64& msg)
{
	ROS_INFO("Received = [%f]", msg.data);
}

int main(int argc, char **argv){
	ros::init(argc, argv, "minimal_subscriber");
	ros::NodeHandle h;
	ros::Subscriber sub = h.subscribe("topic1", 1, callback);

	ros::spin();
	return 0;
}

File: Chapter1(Node_and_Topics)/indigo/scripts/minimal_subscriber.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import Float64

def callback(data):
	rospy.loginfo("Recived = [%f]", data.data)

def listener():
	rospy.init_node("minimal_subscriber")
	rospy.Subscriber("topic1", Float64, callback)
	rospy.spin()

if __name__ == '__main__':
	lr = listener()
